### DISCUSSION TITLE
Release v3.0.0

### DISCUSSION THREAD

**Discussion by detomon** [detomon/wigglebone#15] [Announcements]
## Changes

- A completely new implementation that is frame rate independent.
- Added new `DMWBWiggleRotationModifier3D` and `DMWBWigglePositionModifier3D` nodes to replace the now deprecated `WiggleBone` node.
- The previous behaviour of using the acceleration instead of the velocity for movement turned out to be unstable. Now the velocity is used.

Updated examples can be found in the [examples](https://github.com/detomon/wigglebone/tree/master/examples/wigglebone) directory.

## New features

- The new `spring_freq` property replaces the `stiffness` property. It allows to set an exact oscillation frequency and has an equivalent behaviour to  the old `stiffness` property.
- Support for [free rotating and hanging](https://github.com/detomon/wigglebone#free-rotation) bones (with some [limitations](https://github.com/detomon/wigglebone#limitations)).

---

> [!NOTE]
> The new implementation uses `SkeletonModifier3D` nodes and requires Godot 4.3. A fallback version with an alternative implementation for Godot 4.2 can be found in the [godot-4.2](https://github.com/detomon/wigglebone/tree/godot-4.2) branch.

<hr /><em>This discussion was created from the release <a href='https://github.com/detomon/wigglebone/releases/tag/v3.0.0'>v3.0.0</a>.</em>

**Comment by SchimmelSpreu83** [comment]
Is there a way for me to know, how to convert the old Wiggle Bone values to the new ones?
Because I'm want to use the framerate independent version, but the results are just too different.
And it's very difficult to even come close to the old version.

Anyways, great work 🎉

**Reply by detomon** [reply]
I agree, it is a bit less intuitive to get the values right at first with this implementation.
There is no direct way to reuse old values. Try the following steps and test the behaviour after each:

1. For rotation, set "Linear Scale" to a relatively high value (e.g. 5000).
2. Try to match the old wiggle speed with "Spring Frequency" (more stiffness → higher frequency).
3. Try to match the damping behaviour (how long it takes to come to rest) with "Angular Damp" (rotation) or "Linear Damp" (position). This should match the old behaviour, but with a higher value.
4. For rotation, adjust "Linear Scale" to match the influence when moving the skeleton. Generally it is higher for smaller bones and lower for larger bones.
5. If you are using gravity or other forces, set them last as this can change the spring frequency and it will need to be adjusted again.

There may be certain cases where the behaviour is different and cannot be reproduced.

**Comment by TheRealFame** [comment]
I'll go over a checklist of bone's, And you can use this to help recommend specific bone's for these specific thing's.
(Combination/Rotation/Position)

• Tail - (What Node?)
• Hoodie Strings (What Node?)
• Shoe Laces (What Node?)
• Dread's/Long Hair (What Node?)

Specifically in my case, I don't know what node to use for my tail, So I wanted to ask rather then assume.

**Reply by detomon** [reply]
I don't think there is a right choice. Try both variants and see what looks good and if it matches the behaviour that you want to achieve. For long objects (tail, hair) I would probably use rotation as changing the position can distort the mesh and may not look as good. But it depends on the situation.
